server/data/taskClient.ts

```diff
@@ -35,12 +35,16 @@ export default class TaskClient {
     page: number
     sortDirection: string
     sortBy: TaskSortField
-    taskType?: TaskType
+    taskType?: Array<TaskType>
   }): Promise<PaginatedResponse<Task>> {
+    const filters = {} as Record<string, string>
+    if (taskType.length) {
+      filters.types = taskType.join(',')
+    }
     return this.restClient.getPaginatedResponse({
       path: paths.tasks.index.pattern,
       page: page.toString(),
-      query: { allocatedFilter, allocatedToUserId, apAreaId, sortBy, sortDirection, taskType },
+      query: { ...filters, allocatedFilter, allocatedToUserId, apAreaId, sortBy, sortDirection },
     })
```

Comment:
Rather than adding a `filters` object that just contains one query param and not the rest, perhaps it would be cleaner to move the `query` object up and then conditionally add `taskType`?

```ts
    const query = { allocatedFilter, allocatedToUserId, apAreaId, sortBy, sortDirection }
    if (taskType.length) {
      query.types = taskType.join(',')
    }
    return this.restClient.getPaginatedResponse({
      path: paths.tasks.index.pattern,
      page: page.toString(),
      query,
    })
```

Reply:
instead of amending the existing one better to create new one for immutability

Reply:
The `filters` variable is mutated in the current implementation, so I'm not clear what the immutability benefit is here?

Reply:
here it is new object   query: { ...filters, allocatedFilter, allocatedToUserId, apAreaId, sortBy, sortDirection } but here we are changing existing if (taskType.length) {
      query.types = taskType.join(',')
    }

Reply:
It's just defined the line above in `const query`, so it's isolated to this method